guides/hack/45-operators/05-pipe-operator.md

```diff
@@ -0,0 +1,62 @@
+The Pipe operator allows for a more concise, fluid syntax for chaining together expressions. It looks like this:
+
+```
+f() |> g($$) |> h(5, $$)
+```
+
+which conceptually means:
+
+```
+h(5, g(f()))
+```
+
+That is, evaluate the left hand side (LHS) of the pipe, then evaluate the RHS of the pipe with the result of the LHS bound to the placeholder `$$`.
+
+Runtime execution is explicitly left-to-right - a pipe's LHS is always executed before its RHS. More precisely, the LHS of a Pipe is evaluated and its result stored in a temporary, hidden variable. Then evaluate the RHS where `$$` refers to that hidden variable. This means you can use $$ more than once in the RHS but the LHS will still be evaluated only once. For example:
+
+```
+f() |> g1($$) + g2($$) |> h(5, $$)
+```
+
+roughly means:
+
+```
+h(g1(f()) + g2(f()))
+```
+
+except that `f()`:
+- is evaluated only once
+- is evaluated before either `g1` or `g2`.
+
+It may be easier to conceptualize this as a lambda rewrite:
```

Comment:
I can remove the reification.

A map->filter->count example is a good idea and I can add that. I want an additional example because not everyone is convinced using pipes there is actually easier to read than the standard nested calls, and also pipes are useful beyond simply "the collections library is insufficient".